server/src/main/java/io/deephaven/server/partitionedtable/PartitionedTableServiceGrpcImpl.java

```diff
@@ -199,4 +176,84 @@ public void getTable(
                     });
         }
     }
+
+    @TestUseOnly
+    Table lockAndGetConstituents(@NotNull final GetTableRequest request, final Table keyTable,
+            final PartitionedTable partitionedTable) {
+        final boolean requiresLock = keyTable.isRefreshing() || partitionedTable.table().isRefreshing();
+
+        if (requiresLock) {
+            // validate the update graphs are the same
+            final UpdateGraph keyUpdateGraph = keyTable.getUpdateGraph();
+            final UpdateGraph ptUpdateGraph = partitionedTable.table().getUpdateGraph();
+            if (keyUpdateGraph != null && ptUpdateGraph != null && ptUpdateGraph != keyUpdateGraph) {
+                throw Exceptions.statusRuntimeException(Code.INVALID_ARGUMENT,
+                        "Provided key table UpdateGraph is inconsistent with PartitionedTable UpdateGraph");
+            }
+            if (keyUpdateGraph != null) {
+                return keyUpdateGraph.sharedLock()
+                        .computeLocked(() -> getConstituents(request, keyTable, partitionedTable));
+            } else {
+                return ptUpdateGraph.sharedLock()
+                        .computeLocked(() -> getConstituents(request, keyTable, partitionedTable));
+            }
```

Comment:
Use 
`io.deephaven.engine.updategraph.NotificationQueue.Dependency#getUpdateGraph(io.deephaven.engine.updategraph.NotificationQueue.Dependency...)`
or
`io.deephaven.engine.updategraph.NotificationQueue.Dependency#getUpdateGraph(io.deephaven.engine.updategraph.NotificationQueue.Dependency, io.deephaven.engine.updategraph.NotificationQueue.Dependency...)`
to do the validation and get a singular UG.